packages/editor/src/components/rich-text/index.js

```diff
@@ -773,27 +813,29 @@ export class RichText extends Component {
 		// an empty paragraph into another, then also set the selection to the
 		// end.
 		if ( isSelected && ! prevProps.isSelected && ! this.isActive() ) {
-			const record = this.formatToValue( value );
-			const prevRecord = this.formatToValue( prevProps.value );
+			const record = this.formatToValue( value, annotations );
+			const prevRecord = this.formatToValue( prevProps.value, prevProps.annotations );
 			const length = getTextContent( prevRecord ).length;
 			record.start = length;
 			record.end = length;
 			this.applyRecord( record );
 		}
 	}
 
-	formatToValue( value ) {
+	formatToValue( value, annotations = [] ) {
```

Comment:
Cc @aduth. `annotations` here does not return a new array every time with the annotations right? Is there any we can test that it doesn't? If it would by accident, this would have a big performance impact because `memize` won't return the cache. `value` is merely is string, so this method used to be quite simple.

Also right now with `annotations = []` you're returning a new array on every call if there are no annotations. This should be `annotations = EMPTY_ARRAY` instead.

Reply:
`annotations` should point to the same reference unless an annotation is added or removed, because of the `getAnnotationsForRichText` selector.